public/app/plugins/panel/canvas/utils.ts

```diff
@@ -107,28 +106,155 @@ export function onAddItem(sel: SelectableValue<string>, rootLayer: FrameState |
+      const key = `${link.title}/${link.href}`;
+      if (!linkLookup.has(key)) {
+        links.push(link);
+        linkLookup.add(key);
+      }
+    });
+  }
+};
+
+// TODO: This could be refactored a fair amount, ideally the element specific config code should be owned by each element and not in this shared util file
+export function getDataLinks(
+  dimensionContext: DimensionContext,
+  elementOptions: CanvasElementOptions,
+  data: string | undefined
+): LinkModel[] {
+  const panelData = dimensionContext.getPanelData();
   const frames = panelData?.series;
 
   const links: Array<LinkModel<Field>> = [];
   const linkLookup = new Set<string>();
 
+  const elementConfig = elementOptions.config;
+
   frames?.forEach((frame) => {
     const visibleFields = frame.fields.filter((field) => !Boolean(field.config.custom?.hideFrom?.tooltip));
 
-    if (cfg.text?.field && visibleFields.some((f) => getFieldDisplayName(f, frame) === cfg.text?.field)) {
-      const field = visibleFields.filter((field) => getFieldDisplayName(field, frame) === cfg.text?.field)[0];
-      if (field?.getLinks) {
-        const disp = field.display ? field.display(textData) : { text: `${textData}`, numeric: +textData! };
-        field.getLinks({ calculatedValue: disp }).forEach((link) => {
-          const key = `${link.title}/${link.href}`;
-          if (!linkLookup.has(key)) {
-            links.push(link);
-            linkLookup.add(key);
-          }
-        });
-      }
+    // Text config
+    const isTextTiedToFieldData =
+      elementConfig.text?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementConfig.text?.field);
+    const isTextColorTiedToFieldData =
+      elementConfig.color?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementConfig.color?.field);
+
+    // General element config
+    const isElementBackgroundColorTiedToFieldData =
+      elementOptions?.background?.color?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementOptions?.background?.color?.field);
+    const isElementBackgroundImageTiedToFieldData =
+      elementOptions?.background?.image?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementOptions?.background?.image?.field);
+    const isElementBorderColorTiedToFieldData =
+      elementOptions?.border?.color?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementOptions?.border?.color?.field);
+
+    // Icon config
+    const isIconSVGTiedToFieldData =
+      elementConfig.path?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementConfig.path?.field);
+    const isIconColorTiedToFieldData =
+      elementConfig.fill?.field &&
+      visibleFields.some((field) => getFieldDisplayName(field, frame) === elementConfig.fill?.field);
+
+    // Wind turbine config (maybe remove / not support this?)
```

Comment:
weirdly specific :sweat_smile:

Reply:
![image](https://github.com/grafana/grafana/assets/22381771/232cfed0-d12d-4f80-9615-d56eef065af8)

Reply:
your dad joke game is on point. (get it? on point!?)